ChaosMod/Effects/db/Player/PlayerNoMovRandom.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+	Effect by Moxi
+*/
+
+#include <stdafx.h>
+
+static int int_rand_key;
+
+static void OnStart()
+{
+	int_rand_key = g_random.GetRandomInt(0, 3);
+}
+
+static void OnTick()
+{
+	if (int_rand_key == 0) // Disable Up
+	{
+		if (IS_CONTROL_PRESSED(0, 32) == true)
```

Comment:
You could just check with „ if (IS_CONTROL_PRESSED(0, 32))“ (compares to true)
But do you even need those checks? Can’t you just disable them even if they don’t press them? 🤔

Reply:
First thing I tried was : DISABLE_CONTROL_ACTION(0, 32, true)
So basically disable the INPUT_MOVE_UP_ONLY, but that doesn't work. The only way I found to fully disable moving forward was to disable INPUT_MOVE_UD which disable Up and Down at the same time, thus the workaround.

This is a mainly a problem when you walk. No problem when driving because the game is just using INPUT_VEH_ACCELERATE and INPUT_VEH_BRAKE. There is no "INPUT_VEH_AB" that would be used for accelerating and braking. I hope I make sense.

So in conclusion what I found so far : disabling INPUT_MOVE_UP_ONLY doesn't do anything if INPUT_MOVE_UD is still enable.

Thank you for the tip regarding the if!